File: db-4.8.30/examples_java/src/db/repquote/RepQuoteEnvironment.java
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 1997-2009 Oracle.  All rights reserved.
 *
 * $Id$
 */

package db.repquote;

import com.sleepycat.db.*;

/*
 * A simple wrapper class, that facilitates storing some
 * custom information with an Environment object.
 * The information is used by the Replication callback (handleEvent).
 */
public class RepQuoteEnvironment extends Environment
{
    private boolean appFinished;
    private boolean inClientSync;
    private boolean isMaster;

    public RepQuoteEnvironment(final java.io.File host,
        EnvironmentConfig config)
        throws DatabaseException, java.io.FileNotFoundException
    {
        super(host, config);
        appFinished = false;
        inClientSync = false;
        isMaster = false;
    }

    boolean getAppFinished()
    {
        return appFinished;
    }
    public void setAppFinished(boolean appFinished)
    {
        this.appFinished = appFinished;
    }
    boolean getInClientSync()
    {
        return inClientSync;
    }
    public void setInClientSync(boolean inClientSync)
    {
        this.inClientSync = inClientSync;
    }
    boolean getIsMaster()
    {
        return isMaster;
    }
    public void setIsMaster(boolean isMaster)
    {
        this.isMaster = isMaster;
    }
}

File: db-4.8.30/examples_csharp/ex_repquote/RepQuoteEnvironment.cs
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2009 Oracle.  All rights reserved.
 *
 */
using System;
using System.Collections.Generic;
using System.Text;

using BerkeleyDB;

namespace ex_repquote
{
    public class RepQuoteEnvironment
    {
        private bool _appFinished;
        private bool _inClientSync;
        private bool _isMaster;
        public DatabaseEnvironment env;

        public bool AppFinished { 
            get { return _appFinished; }
            set { _appFinished = value; }
        }
        public bool InClientSync { 
            get { return _inClientSync; }
            set { _inClientSync = value; }
        }
        public bool IsMaster { 
            get { return _isMaster; }
            set { _isMaster = value; }
        }

        public static RepQuoteEnvironment Open(string home, DatabaseEnvironmentConfig cfg)
        {
            RepQuoteEnvironment dbEnv = new RepQuoteEnvironment();
            dbEnv.env = DatabaseEnvironment.Open(home, cfg);
            dbEnv._appFinished = false;
            dbEnv._inClientSync = false;
            dbEnv._isMaster = false;
            return dbEnv;
        }
    }
}
